shared-bindings/audiobusio/PDMIn.c

```diff
@@ -53,14 +53,28 @@
 //|   :param bool mono: True when capturing a single channel of audio, captures two channels otherwise
 //|   :param int oversample: Number of single bit samples to decimate into a final sample. Must be divisible by 8
 //|
-//|   Simple record to buffer::
+//|   Record 8-bit unsigned samples to buffer::
 //|
 //|     import audiobusio
 //|     import board
 //|
 //|     # Prep a buffer to record into
 //|     b = bytearray(200)
-//|     with audiobusio.PDMIn(board.MICROPHONE_DATA, board.MICROPHONE_CLOCK) as mic:
+//|     with audiobusio.PDMIn(board.MICROPHONE_CLOCK, board.MICROPHONE_DATA) as mic:
+//|         mic.record(b, len(b))
+//|
+//|   Record 16-bit unsigned samples to buffer::
+//|
+//|     import audiobusio
+//|     import board
+//|
+//|     # Prep a buffer to record into. The array interface doesn't allow for
+//|     # constructing with a set size so we append to it until we have the size
+//|     # we want.
+//|     b = array.array("H")
+//|     for i in range(200):
```

Comment:
Supposedly faster is `b = array.array("H", [0] * 200)`. It should generate less garbage.

`array.array("H", [0]) * 200` is supposed to be even faster but doesn't work in CPy.

https://stackoverflow.com/questions/3214288/what-is-the-fastest-way-to-initialize-an-integer-array-in-python

Reply:
I'm more worried about memory impact than speed. Doing the [0] * 200 inside first creates a list that is then converted to an array.

Reply:
Yah, but will appending to an `array.array` create a new array of size+1? I thought it was contiguous memory.

Reply:
Looks like its len+8 items each time using m_renew which will try to grow in place I believe. https://github.com/adafruit/circuitpython/blob/master/py/objarray.c#L319

Long term we probably want to switch to a multiply of the array `array.array("H", [0]) * 200` but its not supported now.

Reply:
OK, so never mind for now if you want.

Reply:
Documented in #236